### ISSUE TITLE
Enable ability to have models in more than one package (other than the implementation.models package)

### ISSUE THREAD

**Issue by sharathmalladi** [Azure/autorest.java#1647]
Feature request to allow an autorest configuration that will allow us to separate classes into separate packages as follows:

custom-types-subpackage: administration.models
custom-types: <classes in the administration models package>
custom-types-subpackage2: models
custom-types2: <classes in the models package>

**Comment by weidongxu-microsoft** [comment]
@alzimmermsft @srnagar What's your thought?

**Comment by weidongxu-microsoft** [comment]
@sharathmalladi 

In this namespace schema, you seems need to move some client class to `administration` as well?

**Comment by sharathmalladi** [comment]
Yes, I need some to be in models package and some to be in administration.models package. But I also need files in the implementation package that are not public to the user (such as Error classes that are required by both models and administration.models packages). To serve this purpose, we need the ability to have models, administration.models and implementation packages.

**Comment by alzimmermsft** [comment]
Just a quick clarification, is the use case for this with a single Swagger document or multiple Swagger documents directing code generation?

**Comment by sharathmalladi** [comment]
> Just a quick clarification, is the use case for this with a single Swagger document or multiple Swagger documents directing code generation?

Single swagger document in my case.
